server/src/main/java/org/opensearch/common/util/concurrent/ThreadContextStatePropagator.java

```diff
@@ -22,15 +22,19 @@
 public interface ThreadContextStatePropagator {
     /**
      * Returns the list of transient headers that needs to be propagated from current context to new thread context.
-     * @param source current context transient headers
+     *
+     * @param source          current context transient headers
+     * @param isSystemContext if the propagation is for system context.
      * @return the list of transient headers that needs to be propagated from current context to new thread context
      */
-    Map<String, Object> transients(Map<String, Object> source);
+    Map<String, Object> transients(Map<String, Object> source, boolean isSystemContext);
```

Comment:
This is a breaking change on a public API - lets add overloads and defaults so we can backport to 2.x

Reply:
Very good point, I had an impression this experimental API but it is not, @Gaganjuneja could you please work it out with default method?

```
default Map<String, Object> transients(Map<String, Object> source, boolean isSystemContext) {
    return transients(source);
}
```